packages/@aws-cdk/aws-appsync/lib/key.ts

```diff
@@ -0,0 +1,368 @@
+function concatAndDedup<T>(left: T[], right: T[]): T[] {
+  return left.concat(right).filter((elem, index, self) => {
+    return index === self.indexOf(elem);
+  });
+}
+
+/**
+ * Utility class to represent DynamoDB key conditions.
+ */
+abstract class BaseKeyCondition {
```

Comment:
perhaps also move the the utility / non-exported classes into a folder named `private` and reference them. We do it for a few other modules if you need inspiration

Reply:
does centralizing utility classes improve readability? I mean I dont mind moving it but if it's readability we are after I think moving the non-exporting classes to the bottom of the each file would be better because I linked the code together based on their dependencies. 

If moving to `private` is best for consistency then I'll move it :)

Reply:
I'm not particularly concerned about readability. I like the distinction between exported and non-exported classes since there's less likelihood that it'll accidentally get published or exposed (since `private` won't be in `index.ts`). 

Locality in the file is useful if the utility is not used anywhere else and won't be needed elsewhere. I think it's reasonable to move it down after all the exported entities. I don't have a strong position here so would be content as long as we're making a decision consciously.

Reply:
Talked offline and decided that we should move towards moving to utility to prevent accidental exporting.